File: cpp_solutions/53. Maximum Subarray.cpp
//ss
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        
        int ans = nums[0];
        int temp_ans = nums[0];
        
        for(int x=1;x<nums.size();x++){
            temp_ans = max(nums[x], temp_ans + nums[x]);
            ans = max(ans, temp_ans);
        }
        return ans;
        
    }
};

File: python3_solution_set0/53. Maximum Subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        
        current_ans = nums[0]
        max_ans = nums[0]
        for x in range(1,len(nums)):
            current_ans = max(current_ans+nums[x], nums[x])
            max_ans = max(max_ans, current_ans)
            
        return max_ans
            
        
